pkg/http.ts

```diff
@@ -125,6 +125,10 @@ export class HttpClient implements Requester {
     };
 
     this.baseUrl = config.baseUrl.replace(/\/$/, "");
+    const urlRegex = /^https?:\/\/[^\s/$.?#].[^\s]*$/;
```

Comment:
Descriptive comment about what this regex does would be good. I know its somewhat clear now, but we can always make it more clearer.